### PR TITLE
R batch02

### PR DESCRIPTION
**Pull request by croth1**
:tada: ?

### CODE REVIEW

**Comment by johanneskoester** [comment]
@croth1 thanks for your help! Whenever you create a new PR, can you please mention it in issue #2009, including the package names? If you can, please add the PR number behind the package names in my list at the top. Otherwise, it is really hard to avoid duplicate work.

**Comment by croth1** [comment]
@johanneskoester, yes, will do so in the future.

**Comment by croth1** [comment]
@jakirkham, will depending on `xorg-libxt` work instead of `yum` installing? And if so, what is preferred?

**Comment by jakirkham** [comment]
So there are no established rules with the `xorg-*` packages yet. I think the tricky point is integrating them into the packages closer to the base of the stack (e.g. `qt`). Perhaps another tricky point is how best to handle the differences between X11 support on different OSes. There is some discussion on all this going on in PR ( https://github.com/conda-forge/conda-forge-enhancement-proposals/pull/7 ) and some other PRs and issues scatter around. Hoping some more people chime in.

As for what to do with R packages, I would probably defer to @mingwandroid as he is intimately familiar with them. Though it would be good to hear from @pkgw also given his involvement with getting `xorg-*` packages into conda-forge.

**Comment by pkgw** [comment]
@croth1, yes, if you go this route — conda doesn't let you split between "main" and "development" packages (alas), so installing a package like `xorg-libxt` includes both the runtime libraries and the development files. X.org is actually one of the few places with exceptions to those rule of thumb since, for historical reasons, there are various `xorg-*proto` packages that provide only C headers. However in this case there's no `xorg-xt-proto` or anything.

I doubt that differences in the level of X11 support for different OSs are actually that important. Unless upstream has *really* botched their build configuration scripts, the packages that care about graphics backends will default to the sensible ones on their OSs. In particular, building Cairo with X support enables some new features but doesn't change core functionality — the only way Cairo will ever try to use X as a rendering backend is if the application specifically requests to do so.

Anyway, my vision is that conda-forge should provide and always use its own self-contained stack of X libraries. But as @jakirkham said, there isn't consensus on this point as of yet.

**Comment by mingwandroid** [comment]
@pkgw conda-build 3.0 will have support for split-packages (or sub-packages, the term is not decided upon yet).

**Comment by johanneskoester** [comment]
I think for now it should be fine to use system packages for X11. We can move to the conda packages later once the decisions have been made. It is more urgent to get the whole R channel ported.

File: recipes/r-rjson/build.sh [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+
+# R refuses to build packages that mark themselves as Priority: Recommended
+mv DESCRIPTION DESCRIPTION.old
+grep -v '^Priority: ' DESCRIPTION.old > DESCRIPTION
```
Comment (johanneskoester) [inline recipes/r-rjson/build.sh]: Above two lines can be removed. They are only needed for core R packages.

Comment (croth1) [inline recipes/r-rjson/build.sh]: should we remove them from the skeleton?

**Comment by croth1** [comment]
I agree. I'll update my cleanup script when I have time. Deleting all these comments by hand is not a really feasible solution.

File: recipes/r-rjson/build.sh (hunk 1 above)
Comment (johanneskoester) [inline recipes/r-rjson/build.sh]: You can open a PR at conda/conda-build. For now, I use a cleanup script that simply copies over minimal build scripts to each recipe I have created. I will post it here #2009.

**Comment by johanneskoester** [comment]
I will merge this, because the Appveyor failure is just one of the known timeouts. All but one of the three equivalent x64 jobs in Appveyor pass.

**Pull request merged**